packages/react-reconciler/src/ReactFiberContext.js

```diff
@@ -113,19 +117,23 @@ function hasContextChanged(): boolean {
   return didPerformWorkStackCursor.current;
 }
 
-function isContextConsumer(fiber: Fiber): boolean {
-  return fiber.tag === ClassComponent && fiber.type.contextTypes != null;
-}
-
 function isContextProvider(fiber: Fiber): boolean {
-  return fiber.tag === ClassComponent && fiber.type.childContextTypes != null;
-}
-
-function popContextProvider(fiber: Fiber): void {
-  if (!isContextProvider(fiber)) {
-    return;
+  let type;
+  if (fiber.tag === ClassComponent) {
+    type = fiber.type;
+  } else if (fiber.tag === ClassComponentLazy) {
+    type = fiber.type._reactResult;
+    if (type === null) {
```

Comment:
I still don't like this. I don't understand what assumptions you're doing here. At the very least we should never refer to the result without first referring to the status so that the result has a defined type.

When is this called without this being resolved?

Reply:
I thought that maybe it's possible from `renderSubtreeIntoContainer`